### PR TITLE
(PA-6367) Update task_acceptance for Fedora 40 x86_64

### PR DESCRIPTION
**Pull request by shubhamshinde360**
Added fedora-40 to 'latest_platform_list' method so that it considers builds from nightlies with the 'latest' tag since the versioned release package won't be available for fedora-40 yet given its a new platform we're just introducing. 
Also set 'multiple_puppet7_version' to false for the case of fedora-40 because we would only have just one puppet7 release initially.

### CODE REVIEW

**Comment by shubhamshinde360** [comment]
The task_acceptance tests pass locally

`➜  task_spec git:(PA-6367) ✗ bundle exec rake task_acceptance
/Users/shubham.shinde/.rbenv/versions/3.1.0/bin/ruby -I/Users/shubham.shinde/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.13.0/lib:/Users/shubham.shinde/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-support-3.13.1/lib /Users/shubham.shinde/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.13.0/exe/rspec --pattern spec/acceptance/\*\*/\*_spec.rb
/Users/shubham.shinde/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/beaker-rspec-8.2.0/lib/beaker-rspec/helpers/serverspec.rb:39: warning: already initialized constant Module::VALID_OPTIONS_KEYS
/Users/shubham.shinde/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/specinfra-2.89.0/lib/specinfra/configuration.rb:4: warning: previous definition of VALID_OPTIONS_KEYS was here
Beaker::Hypervisor, found some abs boxes to create
Requesting VMs with job_id: shubham.shinde-1719496378873 Will retry for up to an hour.
Waiting 1s (x1) Total: 0 / 1
Waiting 2s (x2) Total: 0 / 1
Waiting 3s (x3) Total: 0 / 1
Waiting 4s (x4) Total: 0 / 1, Ondemand VMPooler: requesting
Waiting 5s (x5) Total: 0 / 1, Ondemand VMPooler: requesting
Waiting 6s (x6) Total: 0 / 1, Ondemand VMPooler: fedora-40-x86_64 (ready: 0, pending: 1)
Waiting 7s (x7) Total: 0 / 1, Ondemand VMPooler: fedora-40-x86_64 (ready: 0, pending: 1)
Waiting 8s (x8) Total: 0 / 1, Ondemand VMPooler: fedora-40-x86_64 (ready: 0, pending: 1)
Waiting 9s (x9) Total: 0 / 1, Ondemand VMPooler: fedora-40-x86_64 (ready: 0, pending: 1)
Waiting 10s (x10) Total: 0 / 1, Ondemand VMPooler: fedora-40-x86_64 (ready: 0, pending: 1)
Waiting 10s (x11) Total: 0 / 1, Ondemand VMPooler: fedora-40-x86_64 (ready: 0, pending: 1)
  Warning: Skipping ip method to ssh to host as its value is not set. Refer to https://github.com/puppetlabs/beaker/tree/master/docs/how_to/ssh_connection_preference.md to remove this warning
verify_host_key: false is deprecated, use :never
  Warning: Skipping ip method to ssh to host as its value is not set. Refer to https://github.com/puppetlabs/beaker/tree/master/docs/how_to/ssh_connection_preference.md to remove this warning
verify_host_key: false is deprecated, use :never
Disabling updates.puppetlabs.com by modifying hosts file to resolve updates to 127.0.0.1 on lamechian-blot.delivery.puppetlabs.net

install task
Installed puppet-agent on lamechian-blot.delivery.puppetlabs.net: success
Ensuring installed puppet-agent on lamechian-blot.delivery.puppetlabs.net: success
Upgraded puppet-agent to puppet8 on lamechian-blot.delivery.puppetlabs.net: success
  works with version and install tasks

Finished in 2 minutes 52.1 seconds (files took 2 minutes 5.5 seconds to load)
1 example, 0 failures

➜  task_spec git:(PA-6367)`

**Comment by mhashizume** [comment]
These changes will conflict with https://github.com/puppetlabs/puppetlabs-puppet_agent/pull/726, can you wait until that PR is merged and rebase?

**Comment by joshcooper** [comment]
Could you update your commit message to explain why this change is being made, as in what's special about `latest_platform_list`? What is the behavior change that occurs when a platform is added to that method?

**Comment by shubhamshinde360** [comment]
Thanks @joshcooper,
I updated the commit message. Please check now.

**Review by joshcooper — APPROVED** [review]

**Pull request merged**
